src/vs/workbench/contrib/chat/common/promptSyntax/contentProviders/filePromptContentsProvider.ts

```diff
@@ -64,14 +64,33 @@ export class FilePromptContentProvider extends PromptContentsProviderBase<FileCh
 		// get the binary stream of the file contents
 		let fileStream;
 		try {
+			// ensure that the referenced URI points to a file before
+			// trying to get a stream for its contents
+			const info = await this.fileService.resolve(this.uri);
+
+			// validate that the cancellation was not yet requested
+			assert(
+				!cancellationToken?.isCancellationRequested,
+				new CancellationError(),
```

Comment:
Creating an error, which involves capturing a stacktrace synchronously, is one of the more costly things one can do in JS. Looks like this isn't a super hot path but it would be good to avoid this pattern (creating an error unnecesarily) proliferating too much.